src/rpcrawtransaction.cpp

```diff
@@ -444,6 +445,14 @@ UniValue createrawtransaction(const UniValue& params, bool fHelp)
 
     CMutableTransaction rawTx = CreateNewContextualCMutableTransaction(
         Params().GetConsensus(), chainActive.Height() + 1);
+    
+    if (NetworkUpgradeActive(chainActive.Height(), Params().GetConsensus(), Consensus::UPGRADE_OVERWINTER)) {
```

Comment:
`chainActive.Height() + 1` so this is consistent with the transaction formulation itself.

In fact, just define `nextHeight = chainActive.Height() + 1` and then use it in `CreateNewContextualCMutableTransaction()`.